pkg/storage/s3/s3.go

```diff
@@ -176,27 +246,33 @@ func (d *driver) StorageExists(cr *opapi.ImageRegistry, modified *bool) (bool, e
 func (d *driver) StorageChanged(cr *opapi.ImageRegistry, modified *bool) bool {
 	var storageChanged bool
 
-	if cr.Status.Storage.State.S3 == nil || cr.Status.Storage.State.S3.Bucket != cr.Spec.Storage.S3.Bucket {
+	if cr.Status.Storage.State.S3 == nil || cr.Spec.Storage.S3 == nil {
+		storageChanged = true
```

Comment:
isn't it only changed if one is nil and one isn't?  if they're both nil, it's unchanged.

Reply:
If they are both nil, don't we need to go ahead and set up some sort of storage?

Reply:
I guess technically this should never be the case, as bootstrap should put *something* into at least the state ...

Reply:
updated in current push

Reply:
This should be sorted out now